### DISCUSSION TITLE
[System.Text.Json] How to create a new array from existing nodes?

### DISCUSSION THREAD

**Discussion by aventus13** [dotnet/runtime#64737] [Q&A] [answered]
I have the following code:

```csharp
var token // property name, type of string
var array // existing array, type of JsonArray

var newArray = new JsonArray();
array.Select(n => n?[token])
    .Where(n => n != null)
    .ToList()
    .ForEach(n => newArray.Add(n!.AsValue()));
```

However, running this throws the following exception
`System.InvalidOperationException : The node already has a parent.`

I'm trying to figure out how to create a new `JsonArray` from existing `JsonNode` elements. I tried creating new `JsonNode` elements "manually" but as far as I can see this is not possible, given that many members in `System.Text.Json` are internal, even the [JsonNode constructor](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs#L37).

**Comment by huoyaoyuan** [comment]
A node isn't supposed to be reference in two trees, or two different positions in tree. You must clone it first before adding to another position.

**Comment by aventus13** [comment, marked as answer]
@huoyaoyuan thanks for your reply. Yes, I'm aware that a node can only have one parent however that doesn't really answer my question. **How** can I clone a node? I'm unable to find any `Clone` method on either the base `JsonNode` nor any derived types such as `JsonArray` or `JsonObject`.

Please note that I'm trying to avoid having to use serialization/deserialization or reflection.

**Reply by huoyaoyuan** [reply]
You can make an api suggestion for `Clone` method then.

**Reply by danmoseley** [reply]
Answered: please use API suggestion template.

**Reply by huoyaoyuan** [reply]
Cloning is covered by #56592.
